flocker/control/_model.py

```diff
@@ -1128,11 +1128,14 @@ def update_node(self, node_state):
         [original_node] = nodes
         updated_node = original_node.evolver()
         for key, value in node_state.items():
-            if value is not None:
```

Comment:
:question: We should add a comment explaining this optimization. So that future maintainers don't revert it.
We should also create a Pyrsistent issue suggesting that `PClass.set` does this check by default and returns `self` if not modified....although I suppose you might be supplying a value which is equal but which is not a reference to the same existing value....in which case the new value should replace the old.